src/white_book.rb

```diff
@@ -35,24 +35,38 @@ def create_accounts_list
 
       @accounts = sheet.map do |nip, account|
         {
-          nip: nip.to_s.tr('^0-9', ''),
           account: account.to_s.tr('^0-9', ''),
+          accountFound: false,
+          company: nil,
           found: false,
-          valid: nil,
-          virtual: false
+          nip: nil,
+          requestId: nil,
+          valid: false,
+          hasVirtual: false,
+          checked: false
         }
       end
 
       @accounts
     end
 
     def create_accounts_data
-      mf_api = MfAPI.new(accounts, @date)
-      accounts_data = mf_api.accounts_data
+      mf_api = MfAPI.new(@date)
+
+      accounts_data = []
+
+      @accounts.each { |subject|
+        if subject[:account] == ""
+          accounts_data.push(nil)
+          next
+        end
```

Comment:
@piotrkabacinski I think it is a good place to use [each_with_object](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-each_with_object)
[nitpick] I think using `do ... end` instead `{...}` when a block is multiline would be a more _Ruby way_ to write it

Reply:
Hm, can you give an example of how to use it in this case? 

I also think simple `map` could be better here:

```Ruby
accounts_data = @accounts.map do |subject|
  return nil if subject[:account] == ""

  account_data = mf_api.account_data(subject[:account])
  JSON.parse(account_data)
end
```

Reply:
@Aritzaranburu fun fact:`return` in map caused an error, it probably stopped the whole Proc? I replaced it with classic condition:

```Ruby
@accounts_data = @accounts.map do |subject|
  if subject[:account] != ""
    account_data = mf_api.account_data(subject[:account])
    JSON.parse(account_data)
  end
end
```

Reply:
I missed it, sorry. Instead of:
```
return nil if subject[:account] == ""
```
this should work
```
next if subject[:account].to_s.strip.empty?
...
account_data = ...
```

Reply:
Also I think I have never seen `if not` in the wild 😬  https://github.com/rubocop-hq/ruby-style-guide#unless-for-negatives

Reply:
@Aritzaranburu but.. why it's bad? it's like `if (!foo) { ... }` 🤔

Reply:
There is a bit of discussion over here https://github.com/rubocop-hq/ruby-style-guide/issues/329. I think this post is one good reason https://github.com/rubocop-hq/ruby-style-guide/issues/329#issuecomment-62769404.

Also I like this summary:
```
in sum,

- if unless looks weird, try to use a simple if
- don't mix and/not/or with &&/!/||
- only use and/not/or for logical comparisons
```